store.go

```diff
@@ -3163,6 +3163,9 @@ func (s *store) ApplyStagedLayer(args ApplyStagedLayerOptions) (*Layer, error) {
 }
 
 func (s *store) CleanupStagedLayer(diffOutput *drivers.DriverWithDifferOutput) error {
+	if diffOutput.TarSplit != nil {
```

Comment:
Many functions that return `drivers.DriverWithDifferOutput` should document that the caller should ensure `Cleanup…` is called eventually … or whatever the rules are.

Reply:
I've added a patch to finally drop `ApplyDiffWithDiffer` and documented the other place where we use it